core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -523,6 +523,7 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
         }
 
         String[] envArray = env.entrySet().stream()
+                .filter(it -> it.getValue() != null)
```

Comment:
Tangential change - noticed that we don't null check the environment variables we pass through, which seems likely to be a bug.